spi/src/main/java/org/eclipse/dataspaceconnector/spi/asset/AssetIndexLoader.java

```diff
@@ -0,0 +1,14 @@
+package org.eclipse.dataspaceconnector.spi.asset;
+
+import org.eclipse.dataspaceconnector.spi.types.domain.asset.Asset;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.DataAddress;
+
+import java.util.Map;
+
+public interface AssetIndexLoader {
+    String FEATURE = "edc:asset:assetindex:loader";
+
+    void insert(Asset asset, DataAddress address);
+
+    void insertAll(Map<Asset, DataAddress> entries);
```

Comment:
Since `Asset` lacks the `equals` and the `hashCode` overrides, it's quite strange to see a `Map` with `Asset` as key class. I understand that, used in this way, `entries` becomes just a collection of pairs of `Asset` and `DataAddress`, and that's ok, but it's... peculiar. Maybe a `List` of `Map.Entry` could fit better (the implementation should be the same)

Reply:
I agree, a list feels more natural.